applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.cpp

```diff
@@ -669,22 +636,17 @@ void SmallStrainUPwDiffOrderElement::
 
     SizeType Index = 0;
 
-    if ( Dim > 2 ) {
-        for ( SizeType i = 0; i < NumUNodes; ++i ) {
-            rValues[Index++] = rGeom[i].FastGetSolutionStepValue( VELOCITY_X, Step );
-            rValues[Index++] = rGeom[i].FastGetSolutionStepValue( VELOCITY_Y, Step );
-            rValues[Index++] = rGeom[i].FastGetSolutionStepValue( VELOCITY_Z, Step );
-        }
-    }
-    else {
-        for ( SizeType i = 0; i < NumUNodes; ++i ) {
-            rValues[Index++] = rGeom[i].FastGetSolutionStepValue( VELOCITY_X, Step );
-            rValues[Index++] = rGeom[i].FastGetSolutionStepValue( VELOCITY_Y, Step );
+    for ( SizeType i = 0; i < NumUNodes; ++i ) {
+        rValues[Index] = rGeom[i].FastGetSolutionStepValue( VELOCITY_X, Step );
+        ++Index;
+        rValues[Index] = rGeom[i].FastGetSolutionStepValue( VELOCITY_Y, Step );
+        ++Index;
+        if ( Dim > 2 ) {
+            rValues[Index] = rGeom[i].FastGetSolutionStepValue( VELOCITY_Z, Step );
+            ++Index;
         }
     }
-
-    for ( SizeType i = 0; i < NumPNodes; ++i )
-        rValues[Index++] = 0.0;
+    std::fill_n(rValues.begin()+Index, NumPNodes, 0.0);
```

Comment:
Same question here, I like the changes, but is this tested?

Reply:
No, not tested. When I code an error here, no evidence of it shows in the integrationtests